arch/x86_64/src/common/x86_64_acpi.c

```diff
@@ -0,0 +1,667 @@
+
+#include <arch/acpi.h>
+
+#include "x86_64_internal.h"
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#define ACPI_BIOS_MEM_START (0x000e0000)
+#define ACPI_BIOS_MEM_END   (0x000fffff)
+#define ACPI_BIOS_MEM_SIZE  (ACPI_BIOS_MEM_END - ACPI_BIOS_MEM_START)
+#define ACPI_PAGE_FLAGS     (X86_PAGE_PRESENT | X86_PAGE_WR | X86_PAGE_NOCACHE)
+
+/* Debug macros */
+
+#define acpi_info  _info
+#define acpi_err   _err
+#define acpi_warn  _warn
+
+/****************************************************************************
+ * Private Types
+ ****************************************************************************/
+
+struct acpi_s
+{
+  struct acpi_rsdp_s *rsdp;
+  struct acpi_rsdt_s *rsdt;
+  struct acpi_xsdt_s *xsdt;
+  struct acpi_madt_s *madt;
+  struct acpi_mcfg_s *mcfg;
+};
+
+/****************************************************************************
+ * Private Function Prototypes
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+static struct acpi_s g_acpi;
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: acpi_map_region
+ *
+ * Description:
+ *   Map ACPI region.
+ *
+ ****************************************************************************/
+
+static void acpi_map_region(uintptr_t addr, size_t size)
+{
+  /* Map region */
+
+  up_map_region((void *)addr, size, ACPI_PAGE_FLAGS);
+}
+
+/****************************************************************************
+ * Name: acpi_map_rsdt
+ *
+ * Description:
+ *   Map all ACPI tables.
+ *
+ ****************************************************************************/
+
+static void acpi_map_rsdt(void)
+{
+  void                *tps   = NULL;
+  uint32_t            *tp32  = NULL;
+  uint32_t            *end32 = NULL;
+
+  tps = &g_acpi.rsdt->table_ptrs;
+  tp32  = (uint32_t *)tps;
+  end32 = (uint32_t *)((uintptr_t)g_acpi.rsdt + g_acpi.rsdt->sdt.length);
+
+  while (tp32 < end32)
+    {
+      acpi_map_region((uintptr_t)*tp32,
+                      sizeof(struct acpi_sdt_s));
+
+      /* Next table */
+
+      tp32 += 1;
+    }
+}
+
+/****************************************************************************
+ * Name: acpi_map_xsdt
+ *
+ * Description:
+ *   Map all ACPI tables.
+ *
+ ****************************************************************************/
+
+static void acpi_map_xsdt(void)
+{
+  void                *tps   = NULL;
+  uint64_t            *tp64  = NULL;
+  uint64_t            *end64 = NULL;
```

Comment:
Too many spaces here and init is redundant since there is unconditional assignment below